src/test/java/com/github/martynagil/drugstoremanagement/service/ComplaintServiceTest.java

```diff
@@ -3,137 +3,246 @@
+		complaintDoesNotExistsForUpdating();
 
 		assertThrows(
 				EntityNotFoundException.class,
-				() -> complaintService.updateComplaint(complaintUpdateDto, 1L)
+				() -> complaintService.updateComplaint(complaintUpdateToAcceptedDto(), 1L)
 		);
-
 		verify(complaintRepository, never()).save(any());
 	}
 
 	@Test
-	void shouldUpdateWhenComplaintExists() {
-		when(complaintRepository.findById(any()))
+	void shouldUpdateWhenSubmittedComplaintExists() {
+		complaintExistsForUpdating(complaint());
+		Complaint complaint = complaintRepository.findById(1L)
+				.orElseThrow(EntityNotFoundException::new);
+
+		complaintService.updateComplaint(complaintUpdateToAcceptedDto(), 1L);
+
+		assertThat(complaint.getComplaintStatus()).isEqualTo(complaintUpdateToAcceptedDto().getComplaintStatus());
+	}
+
+	@Test
+	void shouldNotUpdateToAcceptedWhenStatusIsRejected() {
+		complaintExistsForUpdating(complaint());
+		complaintService.updateComplaint(complaintUpdateToRejectedDto(), 1L);
+
+		assertThrows(
+				IllegalStatusUpdateException.class,
+				() -> complaintService.updateComplaint(complaintUpdateToAcceptedDto(), 1L)
+		);
+	}
+
+	@Test
+	void shouldNotUpdateToRejectedWhenStatusIsAccepted() {
+		complaintExistsForUpdating(complaint());
+		complaintService.updateComplaint(complaintUpdateToAcceptedDto(), 1L);
+
+		assertThrows(
+				IllegalStatusUpdateException.class,
+				() -> complaintService.updateComplaint(complaintUpdateToRejectedDto(), 1L)
+		);
+	}
+
+	@Test
+	void shouldNotUpdateToSubmittedWhenStatusIsAccepted() {
+		complaintExistsForUpdating(complaint());
+		complaintService.updateComplaint(complaintUpdateToAcceptedDto(), 1L);
+
+		assertThrows(
+				IllegalStatusUpdateException.class,
+				() -> complaintService.updateComplaint(complaintUpdateToSubmittedDto(), 1L)
+		);
+	}
+
+	@Test
+	void shouldNotUpdateToSubmittedWhenStatusIsRejected() {
+		complaintExistsForUpdating(complaint());
+		complaintService.updateComplaint(complaintUpdateToRejectedDto(), 1L);
+
+		assertThrows(
+				IllegalStatusUpdateException.class,
+				() -> complaintService.updateComplaint(complaintUpdateToSubmittedDto(), 1L)
+		);
+	}
+
+	private OngoingStubbing<Optional<Transaction>> findTransactionInRepository() {
+		return when(transactionRepository.findById(any()))
+				.thenReturn(Optional.of(transaction()));
+	}
+
+	private OngoingStubbing<Optional<Product>> findProductInRepository() {
+		return when(productRepository.findById(any()))
+				.thenReturn(Optional.of(product()));
+	}
+
+	private OngoingStubbing<Boolean> complaintDoesNotExistForAdding() {
+		return when(complaintRepository.existsByTransactionIdAndProductIdAndComplaintStatus(any(), any(), any()))
+				.thenReturn(false);
+	}
+
+	private OngoingStubbing<Boolean> complaintExistsForAdding() {
+		return when(complaintRepository.existsByTransactionIdAndProductIdAndComplaintStatus(any(), any(), any()))
+				.thenReturn(true);
+	}
+
+	private OngoingStubbing<Optional<Complaint>> complaintDoesNotExistsForUpdating() {
```

Comment:
po pierwsze, po co z tych metody cokolwiek zwracać?
po drugie, fraza "forUpdating" nic nie wnosi a wprowadza chaos, bo nie rozumiałem o co chodzi czytając testy
po trzecie błąd językowy - does not exist